flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/processor/DynamicFilteringDependencyProcessor.java

```diff
@@ -84,4 +100,108 @@ protected void visitNode(ExecNode<?> node) {
 
         return execGraph;
     }
+
+    private ExecNodeGraph enforceDimSideBlockingExchange(
```

Comment:
I think we can do this after ` entry.getKey().setNeedDynamicFilteringDependency(true);` . We just need to create an Exchange as the directly scan input

Reply:
Not really, the issue is that if we would change the edge from `dim` to `fact source`, `dim`, `fact source` and `join` are still connected via pipeline edges, thus they would still falls into the same region.

Currently we make all the output edges of `dpp node` and its precedent nodes (namely the `calc` and `dim source`) to be blocking. This might be further improved with a complex analysis, but in considering of that currently the dpp node only has two precedent nodes (`calc` and `dim source`) and their outputs are relatively small, we might postpone the optimization in consideration of its complexity.

Reply:
make sense